lib/workers/repository/changelog/index.ts

```diff
@@ -1,21 +1,31 @@
 import pMap from 'p-map';
+import {
+  containsTemplates,
+  exposedConfigOptions,
+} from '../../../util/template';
 import type { BranchUpgradeConfig } from '../../types';
 import { getChangeLogJSON } from '../update/pr/changelog';
 
-// istanbul ignore next
 async function embedChangelog(upgrade: BranchUpgradeConfig): Promise<void> {
+  // getChangeLogJSON returns null on error, so don't try again
+  if (upgrade.logJSON !== undefined) {
+    return;
+  }
   upgrade.logJSON = await getChangeLogJSON(upgrade);
 }
 
-// istanbul ignore next
 export async function embedChangelogs(
-  branchUpgrades: Record<string, BranchUpgradeConfig[]>
+  branches: BranchUpgradeConfig[]
 ): Promise<void> {
-  const upgrades = [];
-  for (const branchName of Object.keys(branchUpgrades)) {
-    for (const upgrade of branchUpgrades[branchName]) {
-      upgrades.push(upgrade);
+  await pMap(branches, embedChangelog, { concurrency: 10 });
+}
+
+export function needsChangelogs(upgrade: BranchUpgradeConfig): boolean {
+  for (const field of exposedConfigOptions) {
+    // fields set by `getChangeLogJSON`
+    if (containsTemplates(upgrade[field], ['logJSON', 'releases'])) {
```

Comment:
What if make `['logJSON', 'releases']` the part of `containsTemplates()`?

Reply:
@rarkins  WDYT?

Would require renaming too, as it's only testing for specific templates

Reply:
🤷

Reply:
so let's leave it here, as it should be removed in a major bump